dipy/nn/synb0.py

```diff
@@ -0,0 +1,417 @@
+            self.dl = tf.keras.layers.Conv3DTranspose(1, kernel_size=1,
+                                                    strides=1, padding='valid')
+
+        def call(self, input):
+            # Encode
+            x = self.ec0(input)
+            syn0 = self.ec1(x)
+
+            x = self.pool0(syn0)
+            x = self.ec2(x)
+            syn1 = self.ec3(x)
+
+            x = self.pool1(syn1)
+            x = self.ec4(x)
+            syn2 = self.ec5(x)
+
+            x = self.pool2(syn2)
+            x = self.ec6(x)
+            x = self.ec7(x)
+
+            # Last layer without relu
+            x = self.el(x)
+
+            x = tf.keras.layers.Concatenate()([self.dc9(x), syn2])
+
+            x = self.dc8(x)
+            x = self.dc7(x)
+
+            x = tf.keras.layers.Concatenate()([self.dc6(x), syn1])
+
+            x = self.dc5(x)
+            x = self.dc4(x)
+
+            x = tf.keras.layers.Concatenate()([self.dc3(x), syn0])
+
+            x = self.dc2(x)
+            x = self.dc1(x)
+
+            x = self.dc0(x)
+
+            # Last layer without relu
+            out = self.dl(x)
+
+            return out
+
+        class encoder_block(tf.keras.layers.Layer):
+            def __init__(self, out_channels, kernel_size, strides, padding):
+                super(Synb0.UNet3D.encoder_block, self).__init__()
+                self.conv3d = tf.keras.layers.Conv3D(out_channels,
+                                                    kernel_size,
+                                                    strides=strides,
+                                                    padding=padding,
+                                                    use_bias=False)
+                self.instnorm = tfa.layers.InstanceNormalization()
+                self.activation = tf.keras.layers.LeakyReLU(0.01)
+
+            def call(self, input):
+                x = self.conv3d(input)
+                x = self.instnorm(x)
+                x = self.activation(x)
+
+                return x
+
+        class decoder_block(tf.keras.layers.Layer):
+            def __init__(self, out_channels, kernel_size, strides, padding):
+                super(Synb0.UNet3D.decoder_block, self).__init__()
+                self.conv3d = tf.keras.layers.Conv3DTranspose(out_channels,
+                                                            kernel_size,
+                                                            strides=strides,
+                                                            padding=padding,
+                                                            use_bias=False)
+                self.instnorm = tfa.layers.InstanceNormalization()
+                self.activation = tf.keras.layers.LeakyReLU(0.01)
+
+            def call(self, input):
+                x = self.conv3d(input)
+                x = self.instnorm(x)
+                x = self.activation(x)
+
+                return x
+
+    def fetch_default_weights(self, idx):
```

Comment:
Could this be idx=0?
What are there 5 choices? Is it ok to always pick the first one?

Reply:
The synb0 pipeline averages the 5 outputs of the 5 trained model weights. While the user can use this function to try just one of the models, it is more intended as an internal class function for the predict function. I can add more description about the function if that helps.